torchaudio/__init__.py

```diff
@@ -338,25 +339,65 @@ def get_sox_bool(i: int = 0) -> Any:
         return _torch_sox.sox_bool(i)
 
 
+_SOX_INITIALIZED = False
+# This variable has a micro lifecycle. (False -> True -> None)
+# False: Not initialized
+# True: Initialized
+# None: Already shut down (should not be initialized again.)
+
+_SOX_SUCCESS_CODE = 0
+# defined at
+# https://fossies.org/dox/sox-14.4.2/sox_8h.html#a8e07e80cebeff3339265d89c387cea93a9ef2b87ec303edfe40751d9a85fadeeb
+
+
 @_audio_backend_guard("sox")
 def initialize_sox() -> int:
```

Comment:
I think the whole logic of `initialize_sox`/`shutdown_sox` can be moved to effect chain and mark them as internal, because users no longer need to worry about it.

Reply:
Essentially, you suggest closing sox on exit only, right? We can discuss that on a separate pull request.